sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -77,28 +76,23 @@ private int onConnectionEvent(final RntbdConnectionEvent event, final Throwable
 
         checkNotNull(exception, "expected non-null exception");
 
-        if (event == RntbdConnectionEvent.READ_EOF) {
-            if (!this.endpoint.isClosed()) {
-
-                if (logger.isDebugEnabled()) {
-                    logger.debug("onConnectionEvent({\"event\":{},\"time\":{},\"endpoint\":{},\"cause\":{})",
-                        event,
-                        RntbdObjectMapper.toJson(Instant.now()),
-                        RntbdObjectMapper.toJson(this.endpoint),
-                        RntbdObjectMapper.toJson(exception));
-                }
-
-                for (Uri addressUri : this.addressUris) {
-                    addressUri.setUnhealthy();
-                }
-
-                return addressUris.size();
+        if (event == RntbdConnectionEvent.READ_EOF || event == RntbdConnectionEvent.READ_FAILURE) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("onConnectionEvent({\"event\":{},\"time\":{},\"endpoint\":{},\"cause\":{})",
+                    event,
+                    RntbdObjectMapper.toJson(Instant.now()),
+                    RntbdObjectMapper.toJson(this.endpoint),
+                    RntbdObjectMapper.toJson(exception));
+            }
 
-            } else {
-                if (logger.isDebugEnabled()) {
-                    logger.debug("Endpoint closed while onConnectionEvent: {}", this.endpoint);
-                }
+            // When idleEndpointTimeout reached, SDK will close all existing channels,
+            // which will translate into ClosedChannelException which does not mean server is in unhealthy status.
+            // But it makes sense to make the server as unhealthy as it is safer to validate the server health again for future requests
+            for (Uri addressUri : this.addressUris) {
+                addressUri.setUnhealthy();
```

Comment:
maybe instead of `Unhealthy`, `RevalidationNeeded` is a better one to use for status

Reply:
+1

Reply:
I will update the name in a different PR - as it is internal implementation details, also the naming discussion may take some time

Reply:
It's not adding new state into the overall state transition right?

Reply:
not really, I am more thinking just name changing: change from `Unhealthy` -> `RevalidationNeeded`